### ISSUE TITLE
For bare metal: CLI should be able to produce cloud-config.yaml file

### ISSUE THREAD

**Issue by faddat** [kontena/kontena#367]
..... since I imagine it is generating them for the other platforms, yeah?

**Comment by jakolehm** [comment]
Yes, it generates them.. How would you use cli to provision bare metal nodes?

**Comment by faddat** [comment]
not for provisioning-- here's how I usually make my coreOS instances:

1) Make a cloud-config.yml
2) integrate it into an .iso file using  https://github.com/jriguera/coreos-usb-creator

So I can't really use it for provisioning, but if it gives me the cloud-config.yaml, then I can be sure that I am making that file without messing anything up.

**Comment by faddat** [comment]
This also creates a dynamic, lightning fast path to Kontena on USB for on premises users.

**Comment by faddat** [comment]
is there any way that I can trick the CLI into giving me that output?

**Comment by miskun** [comment]
Could we have something like this:

```
kontena node custom create [OPTIONS]
--name NAME                   Node name
--version VERSION             Define installed Kontena version (default: "latest")
--output TYPE                 Output configuration file for [ cloud-config | something ] 
```

This way we could auto-generate cloud-config.yml and any other configuration file we might need in the future

**Comment by jakolehm** [comment]
@faddat no tricks for this atm. But I like the idea of cloud-config generator. We have been hitting to this same problem with some private iaas/cloud providers and this surely would help with automation.

**Comment by faddat** [comment]
Closed #172, since resolving this will resolve that.  Further, posting my thoughts on use-cases & what I can contribute in terms of making  this into something fairly unique:

I am having the thought that a chain like this:
 
1. Kontena CLI (passes on cloud-config.yml)
2. https://github.com/jriguera/coreos-usb-creator (passes .iso file) 
3. pixiecore (boots systems)

Make you guys a deal:  Help me out with the cloud-config part (god I hate those things) I'll cook up the rest of it and release that as its own repo (which you could feel free to adopt, or not).  Final item:

 * We're all aware of the hybrid cloud hype, but check this out for a second:  
  * Kontena-cli creates a master for the user in the cloud
  * The script I write connects the users's network to the Kontena Master on a VPN-like connection
  * The user provisions machines by turning them on and selecting that the machines PXE boot, at any number of terra firma (non-cloud) locations

To me, that looks like an angle that no one else has, particularly when you get into the advantages that weave provides for running a network with VPN-like functions.

**Comment by miskun** [comment]
Created separate, more generic issue #461 for implementing the required functionality. Therefore, closing this issue.
